### PR TITLE
Adjust imp.BidFloor using BidAdjustment

### PR DESCRIPTION
**Pull request by Pubmatic-Dhruv-Sonone**
This pr contains changes to adjust the  imp.BidFloor for each bidder request  using the BidAdjustmentFactors

PRD:
https://docs.google.com/document/d/1YAt00RfQaU2Ugciqw1LnirLgjyUz9ibfyjfxWUHR_P0/edit#heading=h.6lovavdxmitt

### CODE REVIEW

**Review by hhhjort — APPROVED** [review]
LGTM

File: exchange/utils.go [hunk 1]
```diff
@@ -93,6 +93,11 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 		}
 	}
 
+	if auctionReq.Account.PriceFloors.IsAdjustForBidAdjustmentEnabled() {
```
Comment (VeronikaSolovei9) [inline exchange/utils.go:96]: Just to confirm if this is needed according to requirements.
You check the boolean flag `AdjustForBidAdjustment bool` first and then check if adjustment factor is present for the bidder: `bidAdjustmentFactors[string(bidderRequest.BidderName)]`. In the requirements document you attached it states: "If ext.prebid.bidadjustmentfactors is specified for any bidder, the feature must be able to adjust the floor as seen by that bidder. " My question is: do we need to check the `AdjustForBidAdjustment` option first?

Comment (Pubmatic-Dhruv-Sonone) [reply exchange/utils.go:96]: @VeronikaSolovei9  ` AdjustForBidAdjustment` is account level configuration and applies to all the bidders  and `bidAdjustmentFactors` is  coming as part of request.  If `AdjustForBidAdjustment` is disabled at account level, then there is no need to adjust  floor for any bidder even if  bidAdjustmentFactors is present in request.   
As per my understanding, PRD says  if bidAdjustmentfactor for  a bidder is present in ext.prebid.bidadjustmentfactors,  update the floor  with that value.

**Review by VeronikaSolovei9 — APPROVED** [review]

**Comment by Pubmatic-Dhruv-Sonone** [comment]
> LGTM

@hhhjort  You approval got dismissed due to last commit. Can you please approve the pr again?

**Pull request merged**